zipline/utils/cache.py

```diff
@@ -334,7 +353,11 @@ def mkdir(self, *path_parts):
             The parts of the path after the working directory.
         """
         path = self.getpath(*path_parts)
-        os.mkdir(path)
+        try:
```

Comment:
I would call this function `ensure_dir` because we have changed the semantics away from `mkdir`. We could have both, where `ensure_dir` is implemented in terms of `mkdir`. What do you think?